### PR TITLE
Initial commit for Intel oneAPI optimized applications

### PR DESCRIPTION
**Pull request by Andrey1994**
The goal of this PR is to create a list of **validated and optimized** applications built with oneAPI compiler. This list will be extended later on.  We want to allow users to install these apps from spack environment and use it as a reference.

### CODE REVIEW

**Comment by rscohn2** [comment]
@eugeneswalker: Could you look this over? Their goal is to eventually publish a build cache with optimized images.

**Comment by eugeneswalker** [comment]
This looks great.

**Review by eugeneswalker — APPROVED** [review]

File: INTEL/CPU/spack.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - hpcg%oneapi +openmp ^intel-oneapi-mpi
+    - hpl@2.3%oneapi +openmp ^intel-oneapi-mkl ^intel-oneapi-mpi
+    - lammps@20230615%oneapi +asphere +class2 +kspace +manybody +misc +molecule +mpiio +opt +replica +rigid +intel +openmp-package ^intel-oneapi-mkl ^intel-oneapi-mpi
+    - openfoam@2206%oneapi cxxflags="-O3 -qopt-zmm-usage=low" ^intel-oneapi-mpi
+    #- openfoam-org@10%oneapi cxxflags="-O3 -qopt-zmm-usage=low" ^intel-oneapi-mpi
+    - stream%oneapi +openmp stream_array_size=500000000 ntimes=500 cflags="-O3 -qopt-zmm-usage=high -mcmodel=large -qopt-streaming-stores=always -fno-builtin -qopt-prefetch-distance=128,16"
+    - wrf@4.4%oneapi ^intel-oneapi-mpi ^parallel-netcdf ^libtirpc@1.2.6 
+  packages:
+    all:
+      require: '%gcc@8.5:'
```
Comment (stephenmsachs) [inline INTEL/CPU/spack.yaml:13]: Is there any reason this needs to be 8.5?

**Comment by stephenmsachs** [comment]
Great addition. Would you be able to add the architecture this is meant for? If not as `target=` then as a comment.

File: INTEL/CPU/spack.yaml (hunk 1 above)
Comment (iermolae) [reply INTEL/CPU/spack.yaml:13]: GCC-8.5 is the least version of GCC we do have in our testing environment. Also it's the latest GCC-8 version released.

**Comment by iermolae** [comment]
> Great addition. Would you be able to add the architecture this is meant for? If not as `target=` then as a comment.

will do

**Comment by Andrey1994** [comment]
I added a comment about targets we tested. In fact it should work on other platforms(both newer and older) and it's mostly depends on application recipe since most of platform specific flags are set there.
Regarding gcc, we set it to >= 8.5 since we didn't test it using older versions, and in theory there can be issues with some apps using older compiler

**Comment by rscohn2** [comment]
Let me know if you are done making changes so I can merge.

**Comment by Andrey1994** [comment]
If there are no other comments from the review, I think we are good right now. We will send PRs with other apps later on

**Pull request merged**